wallet/src/views/wallet/index.jsx

```diff
@@ -14,77 +14,427 @@ import Header from '@Components/Header/header.jsx';
 
 const supportedTokens = importTokens();
 
 const { DEFAULT_ACCOUNT_NUM } = global.config;
 
 const { ethereum } = global;
 
-/**
-This is a modal to detect if a wallet has been initialized
-*/
-
 function WalletModal(props) {
   const [, , deriveAccounts] = useContext(UserContext);
   const [screenMnemonic, setScreenMnemonic] = useState();
+  const [isNewWallet, setIsNewWallet] = useState(true);
+  const [backupFile, setBackupFile] = useState(null);
+  const [isMnemonicValid, setIsMnemonicValid] = useState(false);
+  const [mnemonicRecArray, setMnemonicRecArray] = useState([
+    [
+      { id: 0, word: '' },
+      { id: 1, word: '' },
+      { id: 2, word: '' },
+      { id: 3, word: '' },
+      { id: 4, word: '' },
+      { id: 5, word: '' },
+    ],
+    [
+      { id: 6, word: '' },
+      { id: 7, word: '' },
+      { id: 8, word: '' },
+      { id: 9, word: '' },
+      { id: 10, word: '' },
+      { id: 11, word: '' },
+    ],
+  ]);
+
+  /**
+   *
+   * @param {*} event
+   * @description got the file choosen by the input for upload
+   * file and handle the import commitments and backup flow.
+   */
+  const uploadBackupFile = async event => {
+    event.preventDefault();
+    try {
+      const objectRecovered = await convertFileToObject(event.target.files[0]);
+      setBackupFile(objectRecovered);
+    } catch {
+      setBackupFile(null);
+    }
+  };
+
+  const concatAllWords = () => {
+    return new Promise(resolve => {
+      let mnemonic = '';
+      mnemonicRecArray.forEach((objRow, indexR) => {
+        objRow.forEach((objCol, indexC) => {
+          if (indexR === mnemonicRecArray.length - 1 && indexC === objRow.length - 1) {
+            mnemonic = mnemonic.concat(objCol.word);
+          } else {
+            mnemonic = mnemonic.concat(objCol.word).concat(' ');
+          }
+        });
+      });
+      resolve(mnemonic);
+    });
+  };
+
+  const setIndexedDBObjectsStore = async () => {
+    const commitments = await getIndexedDBObjectRowsFromBackupFile(backupFile, 'commitments');
+    await addObjectStoreToIndexedDB('nightfall_commitments', commitments, 'commitments');
+    const transactions = await getIndexedDBObjectRowsFromBackupFile(backupFile, 'transactions');
+    await addObjectStoreToIndexedDB('nightfall_commitments', transactions, 'transactions');
+  };
+
+  const clearSiteData = indexedDBDatabaseName => {
+    indexedDB.deleteDatabase(indexedDBDatabaseName);
+    window.localStorage.clear();
+    window.sessionStorage.clear();
+  };
+
+  const recoverWallet = async () => {
+    const mnemonic = await concatAllWords();
+    const isValid = validateMnemonic(mnemonic);
+
+    if (!isValid) {
+      setIsMnemonicValid(isValid);
+      return;
+    }
+
+    if (!backupFile) {
+      return;
+    }
+
+    setIsMnemonicValid(isValid);
+    await deriveAccounts(mnemonic, DEFAULT_ACCOUNT_NUM);
+
+    new Promise((resolve, reject) => {
+      let flag = false;
```

Comment:
The code in this promise is so long that I think it deserves a method just for it.